ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/timeline/TimelineComponentView.kt

```diff
@@ -82,39 +92,47 @@ internal fun TimelineComponentView(
                 modifier = Modifier.height(timelineState.itemSpacing.dp)
                     .constrainAs(itemSpacingRef) {
                         top.linkTo(bottomContentBarrier)
-                        if (isLastItem) {
-                            bottom.linkTo(parent.bottom)
-                        }
                     },
             )
 
-            IconComponentView(
-                style = item.icon,
-                state = state,
+            Box(
                 modifier = Modifier.constrainAs(iconRef) {
                     when (timelineState.iconAlignment) {
                         TimelineComponent.IconAlignment.Title -> {
-                            top.linkTo(titleRef.top)
-                            bottom.linkTo(titleRef.bottom)
+                            top.linkTo(currentPreviousItem ?: parent.top)
                             start.linkTo(parent.start)
-                            end.linkTo(titleRef.start)
                         }
                         TimelineComponent.IconAlignment.TitleAndDescription -> {
                             top.linkTo(titleRef.top)
                             bottom.linkTo(descriptionRef.bottom)
                             start.linkTo(parent.start)
-                            end.linkTo(titleRef.start)
                         }
                     }
+                    width = biggestIconWidth?.let { Dimension.value(it) } ?: Dimension.wrapContent
                 },
-            )
+            ) {
+                IconComponentView(
+                    style = item.icon,
+                    state = state,
+                    modifier = Modifier.align(Alignment.TopCenter),
+                )
+            }
 
             TextComponentView(
                 style = item.title,
                 state = state,
                 modifier = Modifier.constrainAs(titleRef) {
-                    top.linkTo(currentPreviousItem ?: parent.top)
-                    start.linkTo(iconRef.end, margin = timelineState.columnGutter.dp)
+                    when (timelineState.iconAlignment) {
+                        TimelineComponent.IconAlignment.Title -> {
+                            centerVerticallyTo(iconRef)
+                        }
+                        TimelineComponent.IconAlignment.TitleAndDescription -> {
+                            top.linkTo(currentPreviousItem ?: parent.top)
+                        }
+                    }
+                    start.linkTo(iconEndBarrier)
+                    end.linkTo(parent.end)
+                    width = Dimension.fillToConstraints
```

Comment:
Without this, the text was increasing more than the size of the parent when the text was longer.